apps/web/server/routers/viewer.tsx

```diff
@@ -940,6 +942,244 @@ const loggedInViewerRouter = createProtectedRouter()
+              if (location.type.includes(integrationQuery)) {
+                return { type: "integrations:daily" };
+              }
+              return location;
+            });
+
+            await prisma.eventType.update({
+              where: {
+                id: eventType.id,
+              },
+              data: {
+                locations: updatedLocations,
+              },
+            });
+          }
+        }
+      }
+
+      // If it's a calendar, remove the destination claendar from the event type
+      if (credential.app?.categories.includes(AppCategories.calendar)) {
+        for (const eventType of eventTypes) {
+          if (eventType.destinationCalendar?.integration === credential.type) {
+            await prisma.destinationCalendar.delete({
+              where: {
+                id: eventType.destinationCalendar.id,
+              },
+            });
+          }
+        }
+      }
+
+      // If it's a payment, hide the event type and set the price to 0. Also cancel all pending bookings
+      if (credential.app?.categories.includes(AppCategories.payment)) {
+        for (const eventType of eventTypes) {
+          if (eventType.price) {
+            await prisma.eventType.update({
+              where: {
+                id: eventType.id,
+              },
+              data: {
+                hidden: true,
+                price: 0,
+              },
+            });
+
+            // Assuming that all bookings under this eventType need to be paid
+            const unpaidBookings = await prisma.booking.findMany({
+              where: {
+                userId: ctx.user.id,
+                eventTypeId: eventType.id,
+                status: "PENDING",
+                paid: false,
+                payment: {
+                  every: {
+                    success: false,
+                  },
+                },
+              },
+              select: {
+                ...bookingMinimalSelect,
+                recurringEventId: true,
+                userId: true,
+                user: {
+                  select: {
+                    id: true,
+                    credentials: true,
+                    email: true,
+                    timeZone: true,
+                    name: true,
+                    destinationCalendar: true,
+                    locale: true,
+                  },
+                },
+                location: true,
+                references: {
+                  select: {
+                    uid: true,
+                    type: true,
+                    externalCalendarId: true,
+                  },
+                },
+                payment: true,
+                paid: true,
+                eventType: {
+                  select: {
+                    recurringEvent: true,
+                    title: true,
+                  },
+                },
+                uid: true,
+                eventTypeId: true,
+                destinationCalendar: true,
```

Comment:
We can abstract this into a new select and use it where we need to use `sendCancelledEmails`

Reply:
We're only filtering for bookings that don't have complete purchases so every booking in this query will be sent an email